docs/utxo-hd/utxo-db/utxo-db-api.tex

```diff
@@ -107,116 +109,119 @@ \subsection{In the consensus layer}
 \label{general-approach}
 
 We wish to deviate as little as possible from the general design of the current
-ledger and consensus layers, particularly their style using pure functions and
-persistent data structures.
+ledger and consensus layers, particularly with respect to using pure functions
+and persistent data structures.
 
 The ledger state will be maintained using a partitioned approach with both
-in-memory and on-disk storage. The large key/value mappings will be kept primarily on
-disk, and all other state kept in memory.
+in-memory and on-disk storage. The large key-value mappings will be kept primarily on
+disk, and all other state will be kept in memory.
 
-We will manipulate the state using pure functions over immutable in-memory data
+We will manipulate the state using pure functions over immutable, in-memory data
 structures. This relies on two main ideas: reading the data into memory in
 advance, and working with differences of data structures.
 
 \subsection{Inspiration from the `anti-caching' database architecture}
 \label{anti-caching}
 
-Our general approach takes inspiration from \cite{anti-caching} who describe
-an OLTP\footnote{OLTP stands for Online Transaction Processing, and is a style
-of database workload. A blockchain ledger database is such a workload.} database architecture that they call `anti-caching'.
+Our general approach takes inspiration from \citet{anti-caching} who describe an
+OLTP\footnote{OLTP stands for Online Transaction Processing, which is a style of
+database workload. A blockchain ledger database would be subjected to such a
+workload.} database architecture that they call `anti-caching'.
 
 The anti-caching architecture reverses the usual notion that a database's value
-is represented on disk, with some of the data cached in memory for efficiency.
-Instead the perspective is that the logical value of the database is represented
-in memory, with much of the data evicted from memory to disk. This is the
-essence of the eponymous anti-cache: the disk is used as an anti-cache to evict
-in-memory data, rather than using memory as a cache for disk data.
-
-%\begin{figure}[h]
-\begin{center}
-\includegraphics{hstore-anticaching-fig1.pdf}
-\begin{quote}
-Diagram by \citet[Figure 1]{anti-caching} illustrating the difference in
-database architecture: ``In (a), the disk is the primary storage for the
-database and data is brought into main memory as it is needed. With the
-anti-caching model in (b), memory is the primary storage and cold data is
-evicted to disk.''
-\end{quote}
-\end{center}
-%\end{figure}
+is represented on disk while some data is cached in memory for efficiency.
+Instead, the perspective is that the logical value of the database is
+represented in memory while much of the data is evicted from memory to disk.
+This is the essence of the eponymous anti-cache: rather than using memory as a
+cache for disk data, the disk is used as an anti-cache to evict in-memory data
```

Comment:
FWIW, I find the previous sentence easier to read

> the disk is used as an anti-cache to evict in-memory data, rather than using memory as a cache for disk data.